pkg/controller/nodeconfig/sync.go

```diff
@@ -250,3 +419,67 @@ func (ncc *Controller) getServiceAccounts() (map[string]*corev1.ServiceAccount,
 
 	return sasMap, nil
 }
+
+func (ncc *Controller) getMatchingNodes(nc *scyllav1alpha1.NodeConfig) ([]*corev1.Node, error) {
```

Comment:
how does this deal with affinities, anti affinities,and the rest of scheduling?

Reply:
I'm not sure I understand - it uses an existing func `IsNodeConfigSelectingNode` which is being used by existing controllers and is covered with unit tests. It covers label selectors, affinities and taints and tolerations that can be set in a NodeConfig spec. Are there any cases not covered by this? If a nodeconfig pod is not scheduled to a node for some reason despite matching these conditions, we should assume unavailable and progressing, no?